pulp_container/app/viewsets.py

```diff
@@ -207,7 +206,99 @@ class ContainerRemoteViewSet(RemoteViewSet):
     }
 
 
-class ContainerRepositoryViewSet(RepositoryViewSet):
+class TagOperationsMixin:
+    """
+    A mixin that adds functionality for creating, deleting, and copying tags.
+    """
+
+    @extend_schema(
+        description="Trigger an asynchronous task to tag an image in the repository",
+        summary="Create a Tag",
+        responses={202: AsyncOperationResponseSerializer},
+        request=serializers.TagImageSerializer,
+    )
+    @action(detail=True, methods=["post"], serializer_class=serializers.TagImageSerializer)
+    def tag(self, request, pk):
+        """
+        Create a task which is responsible for creating a new tag.
+        """
+        repository = self.get_object()
+        request.data["repository"] = repository
+
+        serializer = serializers.TagImageSerializer(data=request.data, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+
+        manifest = serializer.validated_data["manifest"]
+        tag = serializer.validated_data["tag"]
+
+        result = enqueue_with_reservation(
+            tasks.tag_image,
+            [repository, manifest],
+            kwargs={"manifest_pk": manifest.pk, "tag": tag, "repository_pk": repository.pk},
+        )
+        return OperationPostponedResponse(result, request)
+
+    @extend_schema(
+        description="Trigger an asynchronous task to untag an image in the repository",
+        summary="Delete a tag",
+        responses={202: AsyncOperationResponseSerializer},
+        request=serializers.UnTagImageSerializer,
+    )
+    @action(detail=True, methods=["post"], serializer_class=serializers.UnTagImageSerializer)
+    def untag(self, request, pk):
+        """
+        Create a task which is responsible for untagging an image.
+        """
+        repository = self.get_object()
+        request.data["repository"] = repository
+
+        serializer = serializers.UnTagImageSerializer(
+            data=request.data, context={"request": request}
+        )
+        serializer.is_valid(raise_exception=True)
+
+        tag = serializer.validated_data["tag"]
+
+        result = enqueue_with_reservation(
+            tasks.untag_image, [repository], kwargs={"tag": tag, "repository_pk": repository.pk}
+        )
+        return OperationPostponedResponse(result, request)
+
+    @extend_schema(
+        description="Trigger an asynchronous task to copy tags",
+        summary="Copy tags",
+        responses={202: AsyncOperationResponseSerializer},
+        request=serializers.TagCopySerializer,
+    )
+    @action(detail=True, methods=["post"], serializer_class=serializers.TagCopySerializer)
+    def copy_tags(self, request, pk):
```

Comment:
The goal of this story is to enable only /tag and /untag endpoints on the Push Container repo.